eland/operations.py

```diff
@@ -1196,6 +1198,66 @@ def describe(self, query_compiler: "QueryCompiler") -> pd.DataFrame:
             ["count", "mean", "std", "min", "25%", "50%", "75%", "max"]
         )
 
+    def iterrows(
+        self,
+        query_compiler: "QueryCompiler"
+    ) -> Iterable[Tuple[Union[str, Tuple[str, ...]], pd.Series]]:
+        query_params, post_processing = self._resolve_tasks(query_compiler)
+        result_size, sort_params = Operations._query_params_to_size_and_sort(
+            query_params
+        )
+
+        script_fields = query_params.script_fields
+        query = Query(query_params.query)
+
+        body = query.to_search_body()
+        if script_fields is not None:
+            body["script_fields"] = script_fields
+
+        # Only return requested field_names and add them to body
+        _source = query_compiler.get_field_names(include_scripted_fields=False)
+        body["_source"] = _source if _source else False
+
+        if sort_params:
+            body["sort"] = [sort_params]
+
+        es_results_generator = search_yield_hits(
+            query_compiler=query_compiler, body=body, max_number_of_hits=result_size
+        )
+
+        return PandasDataFrameRowsIterator(query_compiler, es_results_generator, post_processing)
+
+    def itertuples(
+        self,
+        query_compiler: "QueryCompiler",
+        index: bool,
+        name: Union[str, None]
+    ) -> Iterable[Tuple[Any, ...]]:
+        query_params, post_processing = self._resolve_tasks(query_compiler)
+        result_size, sort_params = Operations._query_params_to_size_and_sort(
+            query_params
+        )
+
+        script_fields = query_params.script_fields
+        query = Query(query_params.query)
+
+        body = query.to_search_body()
+        if script_fields is not None:
+            body["script_fields"] = script_fields
+
+        # Only return requested field_names and add them to body
+        _source = query_compiler.get_field_names(include_scripted_fields=False)
+        body["_source"] = _source if _source else False
+
+        if sort_params:
+            body["sort"] = [sort_params]
+
+        es_results_generator = search_yield_hits(
+            query_compiler=query_compiler, body=body, max_number_of_hits=result_size
+        )
+
+        return PandasDataFrameTuplesIterator(query_compiler, es_results_generator, post_processing, index, name)
```

Comment:
So, the above method and below have same logic, except for `return` . Can we accomplish it in single method?